File: a_number_after_double_reversal.py
class Solution:
    def isSameAfterReversals(self, num: int) -> bool:
        if num == 0:
            return True
        if str(num)[-1] == '0':
            return False
            
        reversed1 = str(num)[::-1]
        reversed2 = int(reversed1[::-1])

        if reversed2 == num:
            return True

        return False

File: a_number_after_double_reversal.c
bool isSameAfterReversals(int num)
{
    if(num == 0)
    {
        return true;
    }
    
    if(num%10 == 0)
    {
        return false;
    }

    return true;
}

bool isSameAfterReversals(int num)
{
    if(num == 0 || num%10 != 0)
    {
        return true;
    }

    return false;
}